From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sun, 4 Sep 2022 19:26:02 +0200
Subject: Revert clipboard user gesture requirement removal

This reverts commit 4d7b74b051abfe5945f418601fdc2ffc8ce3072c.

License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
---
 .../blink/renderer/modules/clipboard/clipboard_promise.cc   | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/third_party/blink/renderer/modules/clipboard/clipboard_promise.cc b/third_party/blink/renderer/modules/clipboard/clipboard_promise.cc
--- a/third_party/blink/renderer/modules/clipboard/clipboard_promise.cc
+++ b/third_party/blink/renderer/modules/clipboard/clipboard_promise.cc
@@ -543,11 +543,7 @@ void ClipboardPromise::RequestPermission(
       LocalFrame::HasTransientUserActivation(GetLocalFrame());
   base::UmaHistogramBoolean("Blink.Clipboard.HasTransientUserActivation",
                             has_transient_user_activation);
-  // `will_be_sanitized` is false only when we are trying to read/write
-  // web custom formats.
-  if (!will_be_sanitized &&
-      RuntimeEnabledFeatures::ClipboardCustomFormatsEnabled() &&
-      !has_transient_user_activation) {
+  if (!has_transient_user_activation) {
     script_promise_resolver_->Reject(MakeGarbageCollected<DOMException>(
         DOMExceptionCode::kSecurityError,
         "Must be handling a user gesture to use custom clipboard"));
-- 
2.38.1

